src/components/Tool/FixMailingAddresses/Contact.tsx

```diff
@@ -234,33 +206,40 @@ const Contact: React.FC<Props> = ({
                 </Box>
               </Grid>
             </Hidden>
-            {addresses.map((address) => (
-              <Fragment key={address.id}>
-                <Grid item xs={12} md={5} className={classes.paddingB2}>
-                  <Box display="flex" justifyContent="space-between">
-                    <Grid item md={8}>
-                      <Hidden mdUp>
+            {addressesData &&
+              addressesData.map((address) => (
+                <Fragment key={address.id}>
+                  <Grid
+                    item
+                    xs={12}
+                    md={5}
+                    className={classes.paddingB2}
+                    data-testid="address"
+                  >
+                    <Box display="flex" justifyContent="space-between">
+                      <Grid item md={8}>
+                        <Hidden mdUp>
+                          <Typography display="inline">
+                            <strong>{t('Source')}: </strong>
+                          </Typography>
+                        </Hidden>
                         <Typography display="inline">
-                          <strong>{t('Source')}: </strong>
+                          {sourceToStr(t, address.source)}{' '}
                         </Typography>
-                      </Hidden>
-                      <Typography display="inline">
-                        {sourceToStr(t, address.source)}{' '}
-                      </Typography>
-                      <Typography display="inline">
-                        {dateFormatShort(
-                          DateTime.fromISO(
-                            address.startDate || address.createdAt,
-                          ),
-                          locale,
-                        )}
-                      </Typography>
-                    </Grid>
-                    <Grid item md={4} className={classes.alignCenter}>
-                      {!settingPrimaryAddress && (
+                        <Typography display="inline">
+                          {dateFormatShort(
+                            DateTime.fromISO(
+                              address.startDate || address.createdAt,
+                            ),
+                            locale,
+                          )}
+                        </Typography>
+                      </Grid>
+                      <Grid item md={4} className={classes.alignCenter}>
                         <ContactIconContainer
                           aria-label={t('Edit Icon')}
-                          onClick={() => handleSetPrimaryContact(address)}
+                          disabled={!isEditableSource(address.source)}
```

Comment:
Since we use this a few times on the file, lets make this into a variable and use useMemo to cache the value.